rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/CompactView.java

```diff
@@ -553,54 +546,42 @@ private void init() {
             }, ContextMenuEvent.getType());
 
-            this.addMouseOverHandler(new MouseOverHandler() {
-                @Override
-                public void onMouseOver(MouseOverEvent event) {
-                    if (dirty) {
-                        if (globalHover != null)
-                            globalHover.hide();
-                        dirty = false;
-                        if (node != null) {
-                            setHoverNodeLabel();
-                        } else if (host != null) {
-                            hoverLabel.setContents("<strong>Host</strong><br>" + host.getHostName());
-                        } else if (nodesource != null) {
-                            hoverLabel.setContents("<strong>NodeSource</strong><br>" + nodesource.getSourceName());
-                        }
-                        hover.moveTo(event.getClientX() - 155, event.getClientY() - 65);
-                        hover.show();
-                        globalHover = hover;
+            this.addMouseOverHandler(event -> {
+                if (dirty) {
+                    if (globalHover != null)
+                        globalHover.hide();
+                    dirty = false;
+                    if (node != null) {
+                        setHoverNodeLabel();
+                    } else if (host != null) {
+                        hoverLabel.setContents("<strong>Host</strong><br>" + host.getHostName());
+                    } else if (nodesource != null) {
+                        hoverLabel.setContents("<strong>NodeSource</strong><br>" + nodesource.getSourceName());
                     }
+                    hover.moveTo(event.getClientX() - 155, event.getClientY() - 65);
+                    hover.show();
+                    globalHover = hover;
                 }
             });
-            this.addMouseOutHandler(new MouseOutHandler() {
-                @Override
-                public void onMouseOut(MouseOutEvent event) {
-                    dirty = true;
-                    hover.hide();
-                    globalHover = null;
-                }
+            this.addMouseOutHandler(event -> {
+                dirty = true;
+                hover.hide();
+                globalHover = null;
             });
 
-            this.addClickHandler(new ClickHandler() {
-                @Override
-                public void onClick(ClickEvent event) {
-                    _doNotScroll = true;
-                    if (node != null) {
-                        controller.selectNode(node);
-                    } else if (host != null) {
-                        controller.selectHost(host);
-                    } else if (nodesource != null) {
-                        controller.selectNodeSource(nodesource);
-                    }
+            this.addClickHandler(event -> {
+                _doNotScroll = true;
+                if (node != null) {
+                    controller.selectNode(node);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)